src/main/java/seedu/address/storage/DrugSearchUtility.java

```diff
@@ -0,0 +1,126 @@
+
+           String nextRecord[];
+           while ((nextRecord = csvReader.readNext()) != null) {
+               if(nextRecord[1].toLowerCase().contains(keyword.toLowerCase())) {
+                   resultsCache.add(nextRecord);
+               }
+           }
+
+           for(int i = 0; i < resultsCache.size(); i++) {
+               String[] currentRecord = resultsCache.get(i);
+               results = results.concat("\nName: "+currentRecord[1]);
+               results = results.concat("\nActive Ingredient(s): "+currentRecord[10].replace("&&",", "));
+               results = results.concat("\nClassification: "+currentRecord[4]);
+               results = results.concat("\n(for more information, enter \"moreinfo "+(i+1)+"\"");
+               results = results.concat("\n\n");
+           }
+
+           return results;
+        }
+
+        catch(Exception e) {
+            e.printStackTrace();
+            StringWriter s = new StringWriter();
+            PrintWriter p = new PrintWriter(s);
+            e.printStackTrace(p);
+            errorLog.print("--------------------\n"+p.toString()+"\n--------------------\n");
+
+            return "There was an error. Please try again with a different keyword.";
+        }
+    }
+
+    /**
+     * Allows user to read more information about any particular drug that was returned as a search result.
+     * @param index Specifies the drug about which the user wishes to know more.
+     * @return A String containing full pharmacological data about the specified drug.
+     */
+    public static String readMore(int index) {
+        String results = "";
+        if(resultsCache.isEmpty()) {
+            return "Please carry out a search using \"drug [drugname]\" first.";
+        }
+
+        index--;
+        String record[] = resultsCache.get(index);
+        results = results.concat("\nName: "+record[1]);
+        results = results.concat("\nActive Ingredient(s): "+record[10].replace("&&",", "));
+        results = results.concat("\nStrengths Available: "+record[11].replace("&&", ", "));
+        results = results.concat("\nDosage Form: "+record[6]);
+        results = results.concat("\nAdministration : "+record[7].replace("&&", ", "));
+        results = results.concat("\nClassification: "+record[4]);
+        results = results.concat("\nLicense Holder: "+record[2]);
+        results = results.concat("\nATC Code: "+record[5]);
+        results = results.concat("\n\n");
+
+        return results;
+    }
+
+
+    /**
+     * Setter for the FileWriter object that creates the error log file.
+     * Including a setter allows try/catch blocks to be used for exception handling.
+     */
+    private static FileWriter setF() {
+        try {
+            return new FileWriter("./src/main/resources/storage/drugSearchErrorLog.txt", true);
+        } catch (Exception e) {
+            System.out.println("Fatal error: Log file could not be created.");
+            System.exit(-1);
```

Comment:
We shouldn't use system exit in this application - attach your class to the event logger and send it there instead: you'll want to catch the File IO exception and send it as a warning to the logger instead. Avoid using pokemon exception handling as you'll end up catching alot of other things. In the event that FIle IO fails, replace it with a generic empty result.

Reply:
Noted. System.exit was being used just for debugging purposes, unfortunately it seems to have ended up in the final code. Will remove it.